src_c/base.c

```diff
@@ -533,6 +533,103 @@ pg_TwoIntsFromObj(PyObject *obj, int *val1, int *val2)
     return 1;
 }
 
+static int
+pg_IntFromObjEx(PyObject *obj, int *val, const char *msg)
+{
+    if (PyFloat_Check(obj)) {
+        *val = (int)PyFloat_AS_DOUBLE(obj);
+        return 1;
+    }
+
+    *val = PyLong_AsLong(obj);
+    if (PyErr_Occurred()) {
+        if (msg && PyErr_ExceptionMatches(PyExc_TypeError)) {
+            PyErr_SetString(PyExc_TypeError, msg);
+        }
+        return 0;
+    }
+    return 1;
+}
+
+static int
+pg_IntFromSeqIndexEx(PyObject *obj, int index, int *val, const char *msg)
+
+{
+    int result = 0;
+    PyObject *item = PySequence_ITEM(obj, index);
+    if (!item) {
+        return 0;
+    }
+    result = pg_IntFromObjEx(item, val, msg);
+    Py_DECREF(item);
+    return result;
+}
+
+static int
+pg_TwoIntsFromObjEx(PyObject *obj, int *val1, int *val2, const char *msg)
+{
+    Py_ssize_t length;
+    /*Faster path for tuples and lists*/
+    if (PyTuple_Check(obj) || PyList_Check(obj)) {
+        length = PySequence_Fast_GET_SIZE(obj);
```

Comment:
You're taking advantage of the fact that Tuples and Lists are returned verbatim from PySequence_Fast, but that is merely an implementation detail. It would be best to write our code to work up to spec, to support alternative versions or future versions of Python.

Reply:
Actually, i'm taking advantage of the fact that lists/tuples are compatible with sequencefast and i can therefore directly call fast functions.
From what i see this code is already up to spec, as it covers every possibility. With that being said we can't support something that doesn't exist yet, and so you'll always have to open a new PR to modify API to be up to spec. After all this is a start, if something other that tuples or lists will recieve the sequencefast implementation one will add nother check on top and you're done.
Also this implementation after a month+ of experimentation i've found it's the cleanest and most importantly the most  stable and performant one you can get. Goint into the PySequenceFast() rabbit hole conversion is a pain as it's not clear how it handles errors and you can get into trouble with strings as they are convertible sequences. With this you don't have to check for strings as a plus.